SOLIDITY_STYLE_GUIDE.md

```diff
@@ -1851,3 +1853,57 @@ function processStaking(
     ...
 }
 ```
+## Security
+
+### Checks Effects Interactions
+
+Checks effects interactions pattern should be used for writing contract methods which make external calls. This pattern reduces the attack surface for malicious contracts trying to hijack control flow after an external call.
+
+When to use this pattern: 
+
+* It cannot be avoided to hand over control flow to an external entity.
+* You want to protect your functions against re-entrancy attacks.
+
+Details:
+
+1. **Checks**: If input is not acceptable then fail fast and fail hard. 
+
+    ```
+    require(someCondition);
+    
+    ```
+
+2. **Effects**: Optimistically update the contract to a new valid state assuming that interactions will be successful. This protects the contract from re-entrance and race conditions.
+
+    ```
+    balance[msg.sender] = 0;
+    
+    ```
+
+3. **Interactions**: External function calls, `.send()`, `.transfer()` and `.call()` should be done at the end of the method. The return value of external call should be asserted if possible. 
+
```

Comment:
> Regarding L1883, two points:
> 
> 1️⃣ By "asserted" do you mean "required"? Given that the focus is on external calls, I would expect to see `require` and not `assert`.
> 
> I realize that you may be using "asserted" in a more general technical sense, but I think it will be less confusing if we don't overload the term.

You are correct, "asserted" here is general term rather than  solidity `assert`. 

Are you ok with the term `verified` or `checked` ? 
`The return value of external call should be verified if possible.` 
 
> 
> 2️⃣ Do we need to check the result? Given that (a) the external callee cannot be _trusted_ and (b) no state will be modified based on the return value—that would break the pattern—is there really any need to assert/require anything about the return value? I realize this may be what we do in our implemented code, but I'm not sure it's strictly necessary. I would say that the guidance here should either be silent on the matter or recommend assessing the return value only when it seems sensible, and not as often as possible.

The statement says `if possible` so clearly it's not a hard requirement.  For instances like `.send ()`, it doesn't revert transaction on failure, it returns false. 

But maybe I have enforced my personal preference in the guideline on this 😞 
I agree that guideline shouldn't enforce this. 👍 I will remove the sentence:  "The return value of external call should be asserted if possible. "

Reply:
@sarvesh-ost: 👍 to "verified" or "checked"

I agree that "if possible" technically means it is not a hard requirement, but I would also take issue if this guidance recommended doing impossible things 😅 

I support removing the sentence, but I would also be OK with narrower guidance (e.g., "sensible" or "necessary", instead of "possible").

Reply:
"should" already means it's not a hard requirement. It would say "must" if it was a hard requirement. I would opt to remove the "if possible" part.

Reply:
Updated should -> must and removed the sentence.